configure.ac

```diff
@@ -52,14 +52,24 @@ AC_ARG_WITH([nccl],
 AC_ARG_WITH([mpi],
             AC_HELP_STRING([--with-mpi=PATH], [Path to non-standard MPI installation]),
             [AS_IF([test -d $withval/lib64], [mpi_libdir="lib64"], [mpi_libdir="lib"])
-             CFLAGS="-I$withval/include $CFLAGS"
+             MPIINC="$withval/include"
+             CFLAGS="-I$MPIINC $CFLAGS"
```

Comment:
This should really be CPPFLAGS, not CFLAGS.  It matters, because AC_CHECK_HEADERS tries to look with just the preprocessor.